content/articles/2018-07-ggplot2-3-0-0.Rmarkdown

```diff
@@ -0,0 +1,162 @@
+
+### sf
+
+Thanks to the help of [Edzer Pebesma](https://github.com/edzer) and the [r-spatial](https://github.com/r-spatial) team, ggplot2 now has full support for simple features through 
+[sf](https://r-spatial.github.io/sf/) using `geom_sf()` and `coord_sf()`; it now 
+automatically aligns CRS across layers, sets up the correct aspect ratio, and 
+draws a graticule.
+
+```{r sf, message = FALSE}
+nc <- sf::st_read(system.file("shape/nc.shp", package = "sf"))
+ggplot(nc) +
+  geom_sf() +
+  annotate("point", x = -80, y = 35, colour = "red", size = 4)
+```
+
+### Calculated aesthetics
+
+The new `stat()` function offers a cleaner, and better-documented syntax for 
+calculated-aesthetic variables. This replaces the older approach of surrounding 
+the variable name with `...`. Instead of using `aes(y = ..count..)`, you can use 
+`aes(y = stat(count))`. 
+
+This is particularly nice for more complex calculations, as `stat()` only needs 
+to be specified once; e.g. `aes(y = stat(count / max(count)))` rather than 
+`aes(y = ..count.. / max(..count..))`.
+
+### Tag
+
+In addition to title, subtitle, and caption, a new tag label has been added, for identifying plots. Add a tag with `labs(tag = "A")`, style it with the `plot.tag` theme element, and control position with the `plot.tag.position` theme.
+
+```{r tag}
+ggplot(mtcars) +
+  geom_point(aes(disp, mpg)) + 
+  labs(tag = 'A', title = 'Title of this plot')
+```
+
+### Layers: geoms, stats, and position adjustments
+
+The new [`position_dodge2()`](https://ggplot2.tidyverse.org/reference/position_dodge.html) allows you to arrange the horizontal position of plots with variable widths (i.e. bars and rectangles, in addition to box plots). By default, `position_dodge2()` preserves the width of each element. You can choose to preserve the total width by setting the `preserve` argument to `"total"`. Thank you to [Kara Woo](https://github.com/karawoo) for all of her work on this.
+
+```{r dodge-st, fig.show = "hold", out.width = "50%"}
+ggplot(mtcars, aes(factor(cyl), fill = factor(vs))) +
+  geom_bar(position = position_dodge2(preserve = "single"))
+
+ggplot(mtcars, aes(factor(cyl), fill = factor(vs))) +
+  geom_bar(position = position_dodge2(preserve = "total"))
+```
+
+### Scales and guides
+
+Several new functions have been added to make it easy to use Viridis colour 
+scales: `scale_colour_viridis_c()` and `scale_fill_viridis_c()` for continuous, 
+and `scale_colour_viridis_d()` and `scale_fill_viridis_d()` for discrete. 
+Viridis is also now used as the default colour and fill scale for ordered 
+factors.
+
+```{r viridis}
+dsamp <- diamonds[sample(nrow(diamonds), 1000), ]
+d <- ggplot(dsamp, aes(carat, price)) +
+  geom_point(aes(colour = clarity))
+d + scale_colour_viridis_d()
```

Comment:
`scale_colour_continuous()` and `scale_colour_gradient()` are now controlled  by global options `ggplot2.continuous.colour` and `ggplot2.continuous.fill`. You can set them to `"viridis"` to use the viridis colour scale by default:

```R
options(
  ggplot2.continuous.colour = "viridis",
  ggplot2.continuous.full = "viridis"
)
```